src/main/java/cz/muni/fi/pa165/library/services/SingleLoanService.java

```diff
@@ -8,25 +8,51 @@
 
 /** @author Martin Páleník 359817
  *  based on http://zetcode.com/springboot/crudrepository/
+ *
+ *  This represents the business logic of the
+ *  loans of book or books by a member.
  */
 
 @Service
 public class SingleLoanService {
+
     private SingleLoanRepository singleLoanRepository;
 
+    /**
+     * Constructor creates single loan between a member and a book
+     * @param singleLoanRepository
+     */
     public SingleLoanService(SingleLoanRepository singleLoanRepository) {
         this.singleLoanRepository = singleLoanRepository;
     }
 
+    /**
+     * Returns all loans, including the ones
+     * where the book has been already returned.
+     *
+     * @return list of all loans
+     */
     public List<SingleLoan> findAll() {
         return (List<SingleLoan>) singleLoanRepository.findAll();
     }
 
+    /**
+     * Returns number of loans (including past loans)
+     * that has ever been entered in the system.
+     *
+     * @return number of all loans
+     */
     public Long count() {
         return singleLoanRepository.count();
     }
 
+    /**
+     * Deletes a specific SingleLoan with the given "id"
+     * @param userId is ID of SingleLoan we are looking for
+     * @throws IllegalArgumentException if "id" is less than 0
```

Comment:
I think it throws whatever CrudRepostiry throws. Did you specify the exception somewhere?

Reply:
According to [the documentation](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#deleteById-ID-) it throws [IllegalArgumentException](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html?is-external=true), which is an [unchecked exception](https://stackoverflow.com/a/5304151/1459339). Therefore, it does not have to be specified (throws) anywhere. Still, it is thrown by my method.

I don't see what's wrong.